### ISSUE TITLE
Incorrect DiagnosticInfoIgnoringInvalidDebugMetadata Diagnostic Kind

### ISSUE THREAD

**Issue by axelmamode** [llvm/llvm-project#205340]
While trying to use [DiagnosticInfoIgnoringInvalidDebugMetadata](https://llvm.org/doxygen/DiagnosticInfo_8h_source.html#l00248)

I hit this:
```
DiagnosticInfoIgnoringInvalidDebugMetadata(
      const Module &M, DiagnosticSeverity Severity = DS_Warning)
      : DiagnosticInfo(DK_DebugMetadataVersion, Severity), M(M) {}
```
I would have expected that first param to be DK_DebugMetadataInvalid instead of DK_DebugMetadataVersion.

**Comment by rnk** [comment]
I have a basic fix. I looked at how this class is used, and there are no casts to it, so there's no way to observe this bug it seems.

**Comment by axelmamode** [comment]
For what it's worth, this can be observed through DiagnosticInfo::getKind() in the context of using (e.g.) llvm::DiagnosticHandler::DiagHandlerCallback
